cpp/monitoring/gauge.h

```diff
@@ -18,22 +21,63 @@ class Gauge : public Metric {
       const typename NameType<LabelTypes>::name&... label_names,
       const std::string& help);
 
-  virtual double Get(const LabelTypes&...) const = 0;
```

Comment:
It used to be that you'd subclass `Counter`/`Gauge` for your new concrete (e.g. Prometheus) implementation, but I've canned that now and there's just a single concrete `Counter` and `Gauge` impl and the fancy pants magic Prometheus/... bits are in their own exporters.